kola/tests/systemd/sysext.go

```diff
@@ -11,9 +11,178 @@ import (
 	"github.com/flatcar/mantle/kola"
 	"github.com/flatcar/mantle/kola/cluster"
 	"github.com/flatcar/mantle/kola/register"
+	"github.com/flatcar/mantle/kola/tests/util"
 	"github.com/flatcar/mantle/platform/conf"
 )
 
+type (
+	configTemplateParameters struct {
+		DownloadLibraryBase64Contents    string
+		DevContainerScriptBase64Contents string
+		MainScriptBase64Contents         string
+		CheckScriptBase64Contents        string
+	}
+)
+
+var (
+	devContainerScript = util.TrimLeftSpace(`
+#!/bin/bash
+
+set -x
+
+set -euo pipefail
+
+version=$(source /etc/release; echo "${VERSION}")
+version_id=$(source /etc/release; echo "${VERSION_ID}")
+board=$(source /usr/share/flatcar/release; echo "${FLATCAR_RELEASE_BOARD}")
+
+line='SYSEXT_LEVEL=1.0'
+if [[ "${1:-}" = 'version-id' ]]; then line="VERSION_ID=${version_id}"; fi
+
+mkdir -p /work/sysext_rootfs/usr/share/flatcar-sysext-kola-test
+echo "${version_id}" >/work/sysext_rootfs/usr/share/flatcar-sysext-kola-test/file
+mkdir -p /work/sysext_rootfs/usr/lib/extension-release.d
+sysext_arch=x86_64
+if [[ "${board}" = 'arm64-usr' ]]; then sysext_arch=arm64; fi
+metadata=(
+    'ID=flatcar'
+    "${line}"
+    "ARCHITECTURE=${sysext_arch}"
+)
+metadata_file=/work/sysext_rootfs/usr/lib/extension-release.d/extension-release.oem-test
+printf '%s\n' "${metadata[@]}" >"${metadata_file}"
+mksquashfs /work/sysext_rootfs "/work/oem-test-${version}.raw" -all-root
+`)
+
+	mainScript = util.TrimLeftSpace(`
+#!/bin/bash
+
+set -x
+
+set -euo pipefail
+
+source /home/core/download-library.sh
+
+download_dev_container_image flatcar_developer_container.bin
```

Comment:
Using mkosi/systemd-repart would be interesting for a dm-verity signed sysext. I'm fine with keeping to use the devcontainer but maybe it won't work well because in nspawn one can't easily use loopback devices, or?